src/osvr/Util/Logger.cpp

```diff
@@ -131,95 +139,90 @@ namespace util {
         }
 
         LogLevel Logger::getLogLevel() const {
-            return logger_ ? convertFromLevelEnum(logger_->level())
-                           : DEFAULT_LEVEL;
+            return convertFromLevelEnum(logger_->level());
         }
 
         void Logger::setLogLevel(LogLevel level) {
-            if (logger_) {
-                logger_->set_level(convertToLevelEnum(level));
-            }
+            logger_->set_level(convertToLevelEnum(level));
         }
 
         void Logger::flushOn(LogLevel level) {
-            if (logger_) {
-                logger_->flush_on(convertToLevelEnum(level));
-            }
+            logger_->flush_on(convertToLevelEnum(level));
         }
 
-        detail::LineLogger Logger::trace(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->trace(msg)}
-                           : detail::LineLogger{name_, LogLevel::trace, msg};
+        Logger::StreamProxy Logger::trace(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::debug(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->debug(msg)}
-                           : detail::LineLogger{name_, LogLevel::debug, msg};
+        Logger::StreamProxy Logger::debug(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
```

Comment:
That's a good idea. This comment also drew my attention to a horrible series of copy-paste errors.